### PR TITLE
Deprecate deny lists

### PR DESCRIPTION
**Pull request by claire153**

### CODE REVIEW

File: __tests__/summary.test.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import {expect, jest, test} from '@jest/globals'
-import {Change, Changes, ConfigurationOptions, Scorecard} from '../src/schemas'
+import {Changes, ConfigurationOptions, Scorecard} from '../src/schemas'
```
[the remark this answers is not in the document]
Comment (dangoor) [reply __tests__/summary.test.ts:2]: I don't think this is necessary

File: src/summary.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ export function addSummaryToSummary(
   scorecard: Scorecard,
   config: ConfigurationOptions
 ): string {
+  addDenyListsDeprecationWarningToSummary()
```
[the remark this answers is not in the document]
Comment (dangoor) [reply src/summary.ts:25]: This _is_ a good suggestion. You could also add a test case to check that the deprecation warning is only included when deny_licenses is present.

[the remark this answers is not in the document]
Comment (claire153) [reply src/summary.ts:25]: Sounds good. Updated the PR description to reflect that.

File: src/summary.ts [hunk 3]
```diff
@@ -106,6 +108,13 @@ export function addSummaryToSummary(
   return out.join('\n')
 }
 
+function addDenyListsDeprecationWarningToSummary(): void {
+  core.summary.addRaw(
+    `${icons.warning} <strong>Deprecation Warning</strong>: The <em>deny-licenses</em> option is deprecated and will be removed in a future version, use <em>allow-licenses</em> instead.`,
```
[the remark this answers is not in the document]
Comment (claire153) [reply src/summary.ts:113]: Good call. Linked the public issue.

[the remark this answers is not in the document]
Comment (ahpook) [reply src/summary.ts:113]:
```suggestion
    `${icons.warning} <strong>Deprecation Warning</strong>: The <em>deny-licenses</em> option is deprecated and will be removed in the next major version release, use <em>allow-licenses</em> instead.`,
```

File: README.md [hunk 4]
```diff
@@ -106,7 +105,7 @@ All configuration options are optional.
...
-| `deny-licenses`\*                      | Contains a list of prohibited licenses. The action will fail on pull requests that introduce dependencies with licenses that match the list.                                                                                              | Any [SPDX-compliant identifier(s)](https://spdx.org/licenses/)                                               | none          |
+| `deny-licenses`\*                      | ⚠️ Deprecated and will be removed in a future version. <br> Contains a list of prohibited licenses. The action will fail on pull requests that introduce dependencies with licenses that match the list.                                  | Any [SPDX-compliant identifier(s)](https://spdx.org/licenses/)                                               | none          |
```
[the remark this answers is not in the document]
Comment (ahpook) [reply README.md:108]: Same as above, this should reference the next major version bump.

File: __tests__/summary.test.ts [hunk 5]
```diff
@@ -119,8 +119,25 @@ test('adds deprecation warning for deny-licenses option', () => {
   )
   const text = core.summary.stringify()
 
+  expect(text).not.toContain(
+    '⚠️ <strong>Deprecation Warning</strong>: The <em>deny-licenses</em> option is deprecated and will be removed in the next major version release, use <em>allow-licenses</em> instead.'
```
Comment (dangoor) [inline __tests__/summary.test.ts:123]: nit: For a `not.toContain` test, I'd check for something short like `deny-licenses` or `Deprecation Warning` because any little text change would make this test always pass and not tell us anything useful.

Comment (dangoor) [reply __tests__/summary.test.ts:123]: Actually, looks like this is already the case… the deprecation warning _could_ be there, but the text of the warning is different from what we're testing for.

File: __tests__/summary.test.ts [hunk 6]
```diff
@@ -119,8 +119,25 @@ test('adds deprecation warning for deny-licenses option', () => {
...
+    emptyInvalidLicenseChanges,
+    emptyChanges,
+    scorecard,
+    config
+  )
+  const text = core.summary.stringify()
+
   expect(text).toContain(
-    '⚠️ <strong>Deprecation Warning</strong>: The <em>deny-licenses</em> option is deprecated and will be removed in a future version, use <em>allow-licenses</em> instead.'
+    '⚠️ <strong>Deprecation Warning</strong>: The <em>deny-licenses</em> option is deprecated for possible removal in the next major release. See [Deprecate the deny-licenses option #938](https://github.com/actions/dependency-review-action/issues/938) for more information.'
```
Comment (dangoor) [inline __tests__/summary.test.ts:140]: This one is more personal preference, but I would make this test resilient to change just like in the negative case. I'd rather not see this test have to change every time there's a minor text edit.

**Review by dangoor — COMMENTED** [review]
I don't have a strong preference on the warning suppression, so I'll leave that to @ahpook. From my perspective, a little tweak to the tests and this looks good!

**Comment by ahpook** [comment]
> Regarding your second point about adding an option to suppress deprecation messages: the warning only appears in the summary as required by the task (see screenshot in this PR's description).

Ah ok, that's fine. I'd thought it was inline in the PR.

**Review by ahpook — APPROVED** [review]
Nice, i like it!
